src/core/rebalance/src/4C_rebalance_graph_based.cpp

```diff
@@ -301,7 +301,8 @@ std::shared_ptr<const Core::LinAlg::Graph> Core::Rebalance::build_graph(
     }
     Core::Communication::broadcast(&size, 1, proc, dis.get_comm());
     if (proc != myrank) recvnodes.resize(size);
-    Core::Communication::broadcast(&recvnodes[0], size, proc, dis.get_comm());
+    Core::Communication::broadcast(
+        recvnodes.empty() ? nullptr : &recvnodes[0], size, proc, dis.get_comm());
```

Comment:
Is it allowed to broadcast the nullptr?


```suggestion
    if (!recvnodes.empty()) Core::Communication::broadcast(&recvnodes[0], size, proc, dis.get_comm());
```